### PR TITLE
feat(useVModel): support `clone` option

### PR DESCRIPTION
**Pull request by edumudu**
### Description

Fixes side effect when using `useVModel` with `passive` flag and with an object as value.
Added a clone option to clone the obj before set to `ref`

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Review by antfu — CHANGES_REQUESTED** [review]
It would be better to be a separate `clone` option that accepts `boolean` and custom functions. Refer to https://github.com/vueuse/vueuse/blob/98104e172cd23a6f212fe2985f8a8036176d80db/packages/core/useManualRefHistory/index.ts#L18-L24

**Comment by edumudu** [comment]
done

File: packages/core/useVModel/index.test.ts [hunk 1]
```diff
@@ -178,7 +178,7 @@ describe('useVModel', () => {
     expect(emitValue instanceof SomeClass).toBeTruthy()
   })
 
-  it('should be side effect free when using objects', async () => {
+  it('should clone object', async () => {
```
Comment (antfu) [inline packages/core/useVModel/index.test.ts:181]: We might better to keep the original test and start as a new one

File: packages/shared/utils/index.ts [hunk 2]
```diff
@@ -104,20 +101,3 @@ export function objectPick<O, T extends keyof O>(obj: O, keys: T[], omitUndefine
     return n
   }, {} as Pick<O, T>)
 }
-
-/**
- * Simple recursive deep clone
- *
- * @category Object
- */
-export const cloneDeep = (obj: any) => {
```
Comment (antfu) [inline packages/shared/utils/index.ts]: Please revert this change

Comment (edumudu) [reply packages/shared/utils/index.ts]: you wanna to use this as default function instead `JSON.parse(JSON.stringfy(...))`?

**Comment by antfu** [comment]
Ah, nvm. it seems I was reviewing the wrong diff. This looks good to me in general but I need to hold it for a while long to make sure it behave consistent with #2045

**Review by antfu — APPROVED** [review]

**Review by antfu — DISMISSED** [review]

**Review by antfu — APPROVED** [review]
